--- crypto/x509v3/x509v3.h	2018-03-27 21:54:46.000000000 +0800
+++ crypto/x509v3/x509v3.h	2018-07-05 14:33:02.000000000 +0800
@@ -688,7 +688,9 @@
                         int ml);
 int X509V3_EXT_print(BIO *out, X509_EXTENSION *ext, unsigned long flag,
                      int indent);
+#ifndef OPENSSL_NO_FP_API
 int X509V3_EXT_print_fp(FILE *out, X509_EXTENSION *ext, int flag, int indent);
+#endif
 
 int X509V3_extensions_print(BIO *out, char *title,
                             STACK_OF(X509_EXTENSION) *exts,
